contents/docs/apps/build/reference.md

```diff
@@ -256,37 +256,33 @@ async function processEvent(event, { config}) {
 
 As you can see, the function receives the event before it is ingested by PostHog, adds properties to it (or modifies them), and returns the enriched event, which will then be ingested by PostHog (after all apps run).
 
-Note that you cannot use storage nor cache nor external calls in processEvent apps in PostHog cloud. Furthermore you can only define one of `processEvent` or `onEvent` per app.
+Note that you cannot use storage nor cache nor external calls in processEvent apps in PostHog cloud. Furthermore you can only define one of `processEvent` or `composeWebhook` per app.
 
-## `onEvent` function
+## `composeWebhook` function
 
-> **Minimum PostHog version:** 1.25.0 
+> **Minimum PostHog hash:** https://github.com/PostHog/posthog/commit/0137b9d40d8c0b4a7183fd6bb3c718a35d116b95
 
-`onEvent` works similarly to `processEvent`, except any returned value is ignored by the app server. In other words, `onEvent` can read an event but not modify it. 
-
-In addition, `onEvent` functions will run after all enabled apps have run `processEvent`. This ensures you will be receiving an event following all possible modifications to it.
-
-This was originally built for and is particularly useful for export apps. These apps need to receive the "final form" of an event and send it out of PostHog, without having to modify it.
+`composeWebhook` is a non-async function that returns `Webhook` object. This is called at the end of the pipeline after the event has been stored in PostHog data store. It allows users to send data out of PostHog one event at a time. See batch exports for exporting events in batches. The function can also return null if for this event we don't want to send the webhook.
```

Comment:
I would also reword this a little bit:

```suggestion
`composeWebhook` is a non-async function that returns `Webhook` object. This object triggers an HTTP request to the `url` defined in it at the end of the pipeline after the event has been stored in PostHog data store. It allows users to submit data to their own HTTP endpoint when an event happens. The function can also return null if for a specific event we don't want to trigger an HTTP request. If you are looking to export event data from PostHog, look into [batch exports](/docs/cdp/batch-exports).
```

Reply:
The reasoning is that the existing definition kind of makes it seem like Webhooks are a tool to export event data: Users can potentially write anything in the webhook payload, or nothing at all, so I preferred a more generic definition, and leaned more into the "triggered by an event" aspect of a Webhook.

Reply:
how about this:
```suggestion
`composeWebhook` is a non-async function that is executed at the end of the pipeline. It allows users to submit data to their own HTTP endpoint when an event happens. The function can return 
- null if for a specific event we don't want to trigger an HTTP request. 
- `Webhook` object, for which we'll trigger an HTTP request to the url with the payload, method and headers returned.
If you are looking to export event data from PostHog, look into [batch exports](/docs/cdp/batch-exports).
```